lib/cocoapods-core/cdn_source.rb

```diff
@@ -56,8 +56,9 @@ def refresh_metadata
     end
 
     def preheat_existing_files
-      all_existing_files = [repo.join('**/*.yml'), repo.join('**/*.txt'), repo.join('**/*.json')].map(&Pathname.method(:glob)).flatten
-      loaders = all_existing_files.map { |f| f.relative_path_from(repo).to_s }.map do |file|
+      files_to_update = files_definitely_to_update + deprecated_local_podspecs
+      debug "CDN: #{name} Going to update #{files_to_update.count} files"
```

Comment:
This is check for updates, right? Only actually doing the update if the etags don't match?

Reply:
ETag is a server-side feature. You send an `if-none-match`, server either sends you a new file, or replies with `HTTP 304 Not Modified`.